lib/shopify_app/omniauth/omniauth_configuration.rb

```diff
@@ -50,6 +50,8 @@ def default_request_online_tokens?
     end
 
     def update_shop_scopes?
+      ShopifyApp.configuration.shop_access_scopes_strategy.scopes_mismatch?(shop_domain)
+    rescue NotImplementedError
```

Comment:
Ideally we don't use errors for control flow. If we're not using a session store that supports access tokens can we just return `false` in `scopes_mismatch?`?

Reply:
We could do that. Only reason we're not is because if the developers are trying to use the strategies to check scope matches (i.e. in ScopesVerification), they may assume that they have the appropriate scopes when access_scopes aren't actually stored. This basically silently fails for them and can possibly hide the underlying issue for them later on. In cases like OmniAuth and Callback controller, it's okay to fail silently because we want default behaviour.

Reply:
This won't be a problem when we later move onto a major version where the gem expects to have access_scopes by default.

Reply:
See my comment here about a potentially more elegant way to handle this: https://github.com/Shopify/shopify_app/pull/1197/files#r586703216